### PR TITLE
MMS test for ShiftedMetricInterp

### PR DESCRIPTION
**Pull request by johnomotani**
I found this test for parallel derivatives using `ShiftedMetricInterp` in an old branch. Probably worth merging.

### CODE REVIEW

File: tests/MMS/shiftedmetricinterp/runtest [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+import itertools
+from sys import exit
+
+errorlist = []
+
+boutcore.init("-d data -q -q -q".split(" "))
+
+
+def runtests(functions, derivatives, direction, stag, msg, ordertol=0.1):
+    global errorlist
```
Comment (ZedThree) [inline tests/MMS/shiftedmetricinterp/runtest:19]:
```suggestion
def runtests(functions, derivatives, direction, stag, msg, errorlist, ordertol=0.1):
```

File: tests/MMS/shiftedmetricinterp/runtest [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+    ["sin(%s + (z-mesh:zShift))", "cos(%s + (z-mesh:zShift))"],
+    ["cos(%s + (z-mesh:zShift))", "-sin(%s + (z-mesh:zShift))"],
+]
+
+derivatives = [
+    [2, "C2"],
+    # [4,"C4"] ,
+    # [2,"W2"] ,
+    # [3,"W3"] ,
+    # [2,"S2"] ,
```
Comment (ZedThree) [inline tests/MMS/shiftedmetricinterp/runtest:134]:
```suggestion
```

File: tests/MMS/shiftedmetricinterp/runtest [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+]
+
+direction = ["y", "1", 2, boutcore.DDY]
+
+runtests(functions, derivatives, direction, stag=False, msg="DD")
+
+derivatives = [
+    [2, "C2"],
+    # [4,"C4"],
```
Comment (ZedThree) [inline tests/MMS/shiftedmetricinterp/runtest:143]:
```suggestion
```

File: tests/MMS/shiftedmetricinterp/runtest [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+]
+
+# runtests(functions,derivatives,direction,stag=True,msg="DD")
```
Comment (ZedThree) [inline tests/MMS/shiftedmetricinterp/runtest:146]:
```suggestion
```

File: tests/MMS/shiftedmetricinterp/runtest [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+functions = [
+    ["sin(%s + (z-mesh:zShift))", "-sin(%s + (z-mesh:zShift))"],
+    ["cos(%s + (z-mesh:zShift))", "-cos(%s + (z-mesh:zShift))"],
+]
+
+derivatives = [
+    [2, "C2"],
+    # [4,"C4"]
+]
+
```
Comment (ZedThree) [inline tests/MMS/shiftedmetricinterp/runtest:158]:
```suggestion
```

File: tests/MMS/shiftedmetricinterp/runtest [hunk 6]
```diff
@@ -0,0 +1,176 @@
...
+runtests(functions, derivatives, direction, False, "D2D2", ordertol=0.2)
+
+derivatives = [
+    [2, "C2"],
+]
+
+# runtests(functions,derivatives,direction,True,"D2D2")
+
+
```
Comment (ZedThree) [inline tests/MMS/shiftedmetricinterp/runtest:169]:
```suggestion
```

File: tests/MMS/shiftedmetricinterp/runtest [hunk 7]
```diff
@@ -0,0 +1,176 @@
...
+    [2, "C2"],
+    # [4,"C4"] ,
+    # [2,"W2"] ,
+    # [3,"W3"] ,
+    # [2,"S2"] ,
+]
+
+direction = ["y", "1", 2, boutcore.DDY]
+
+runtests(functions, derivatives, direction, stag=False, msg="DD")
```
Comment (johnomotani) [inline tests/MMS/shiftedmetricinterp/runtest:139]:
```suggestion
runtests(functions, derivatives, direction, stag=False, msg="DD", errorlist)
```

File: tests/MMS/shiftedmetricinterp/runtest [hunk 8]
```diff
@@ -0,0 +1,176 @@
...
+]
+
+derivatives = [
+    [2, "C2"],
+    # [4,"C4"]
+]
+
+direction = ["y", "1", 2, boutcore.D2DY2]
+
+runtests(functions, derivatives, direction, False, "D2D2", ordertol=0.2)
```
Comment (johnomotani) [inline tests/MMS/shiftedmetricinterp/runtest:161]:
```suggestion
runtests(functions, derivatives, direction, False, "D2D2", errorlist, ordertol=0.2)
```

**Comment by johnomotani** [comment]
Ready to merge now. The tests have passed - the recent failures were just failed uploads in the coverage tests.

**Pull request merged**